media/js/src/multipleChoiceQuestion2.jsx

```diff
@@ -23,6 +22,37 @@ export const MultipleChoiceQuestion2 = ({ isSubmitted, setIsSubmitted,
         const checkResult = result.length > 0 && !result.includes(false);
         return checkResult;
     };
+    const [submissionId, setSubmissionId] = useState();
+
+    const createSubmission = async() => {
+        // Define the data to be saved based on the plot type
+        const data = {};
+
+        const payload = {
+            simulation: 2,
+            data: data
+        };
+
+        const url = `/course/${coursePk}/api/create-sub/`;
+
+        authedFetch(url, 'POST', payload)
+            .then(response => {
+                if (response.status === 201) {
+                    return response.json();
+                } else {
+                    throw `Error (${response.status}) ${response.statusText}`;
+                }
+            })
+            .then(data => {
+                setSubmissionId(data.submission_id);
+            })
+            .catch(error => {
+                console.error('Error saving graph data:', error);
+                throw error;
+            });
+    };
+
+    useEffect(() => {createSubmission();}, []);
```

Comment:
It looks like `useEffect()` is already defined in this class at line 83 -- maybe this code is meant to go there?

Also I am not clear on the context of this code so I will let @ndittren review, but in general an object creation like a submission would happen on a user action, rather than on pageload (useEffect). But maybe this makes sense in useEffect depending on how things are set up.

Reply:
That's the code for a different `useEffect`. The above code is only supposed to trigger on mount, where the `useEffect` on line 83 triggers every time a change to `isCorrect` is detected.

Reply:
Ah okay - I see now, with the other useEffect triggering on `isCorrect`. I didn't know this is how it's done in the new react style. I would at least keep the different `useEffect` definitions next to each other in the code to keep things organized. https://stackoverflow.com/a/54004148